ci_scripts/test_sync_target_db.py

```diff
@@ -0,0 +1,270 @@
+    {
+        "type": "target",
+        "id": "2",
+        "attributes": {
+            "features": {
+                "mbed_enabled": [
+                    "Advanced"
+                ],
+                "mbed_os_support": [
+                    "Mbed OS 5.10",
+                    "Mbed OS 5.11",
+                    "Mbed OS 5.12",
+                    "Mbed OS 5.13",
+                    "Mbed OS 5.14",
+                    "Mbed OS 5.15",
+                    "Mbed OS 5.8",
+                    "Mbed OS 5.9"
+                ],
+                "antenna": [
+                    "Connector"
+                ],
+                "certification": [
+                    "AS/NZS (Australia and New Zealand)",
+                    "CE (Europe)",
+                    "FCC/CFR (USA)",
+                    "IC RSS (Canada)"
+                ],
+                "communication": [
+                    "LoRa"
+                ],
+                "interface_firmware": [
+                    "DAPLink"
+                ],
+                "target_core": [
+                    "Cortex-M3"
+                ],
+                "mbed_studio_support": [
+                    "Build and run"
+                ]
+            },
+            "board_type": "MTB_MTS_XDOT",
+            "flash_size": 256,
+            "name": "Multitech xDOT",
+            "product_code": "0453",
+            "ram_size": 64,
+            "target_type": "module",
+            "hidden": false,
+            "device_name": "STM32L151CC"
+        }
+    }
+]"""
+
+
+class TestSyncTargetDB(TestCase):
+    def test_diff_objects_from_json_returns_diffs(self):
+        with mock.patch("ci_scripts.sync_target_database.MbedTargets") as mock_online_targets:
+            mock_online_targets.json_dump.return_value = ONLINE_TARGET_DATABASE
+            diffs = sync_target_database.diff_objects_from_json_str(
+                mock_online_targets.json_dump(), OFFLINE_TARGET_DATABASE
+            )
+            self.assertEqual(diffs, json.loads(EXPECTED_DIFF), "Actual diff and expected diff should match")
+
+    def test_diff_objects_from_json_returns_none_when_no_change(self):
+        diffs = sync_target_database.diff_objects_from_json_str(OFFLINE_TARGET_DATABASE, OFFLINE_TARGET_DATABASE)
+        self.assertEqual(diffs, [], "Should return an empty list when no diff")
+
+    def test_news_file_item_text_formatting(self):
+        with mock.patch("ci_scripts.sync_target_database.Path", autospec=True) as mock_path:
+            mock_path().exists.return_value = False
+            text = sync_target_database.create_new_boards_news_item_text(json.loads(ONLINE_TARGET_DATABASE))
+            self.assertEqual(
+                text, "New boards added: MTB_UBLOX_NINA_B1, MTB_MTS_XDOT", "Text should be formatted correctly."
+            )
+
+    def test_news_file_name_suffix_increments_when_path_exists(self):
+        with mock.patch("ci_scripts.sync_target_database.Path", autospec=True) as mock_path:
```

Comment:
Wonder if using something like `pyfakefs` wouldn't make this less implementation dependent (fragile).